Lib/asyncio/unix_events.py

```diff
@@ -460,6 +470,27 @@ def cb(fut):
                     self.remove_writer(fd)
         fut.add_done_callback(cb)
 
+    def _stop_serving(self, sock):
+        # Is this a unix socket that needs cleanup?
+        if sock in self._unix_server_sockets:
+            path = sock.getsockname()
+        else:
+            path = None
+
+        super()._stop_serving(sock)
+
+        if path is not None:
+            prev_ino = self._unix_server_sockets[sock]
+            del self._unix_server_sockets[sock]
+            try:
+                if os.stat(path).st_ino == prev_ino:
+                    os.unlink(path)
```

Comment:
Shouldn't the `if` be outside the `try`? If we're not unlinking, there are no exceptions to catch.

Reply:
`os.stat()` could also throw errors, e.g. `ENOENT`.